### ISSUE TITLE
Ferry Duration Times Inconsistent for Cars, Bicycles and Pedestrian

### ISSUE THREAD

**Issue by cliffordsnow** [graphhopper/graphhopper#1116]
I've been adding in ferry duration times in OSM for ferry routes in Washington State. Testing out the results on Graphhopper, I noticed that the router gives different times on the same ferry route for cars, bicycles and pedestrians. For example the Mukilteo-Clinton Ferry (5901929)1 shows the following results using the graphhopper.com/maps:

* Car 4.42km will take 44min
* Bicycle 4.42km will take 26min
* Pedestrian 4.42km will take 29min

The actual duration of the ferry route is 20 minutes which should be the same for all modes of transportation.

**Comment by boldtrn** [comment]
Thanks for bringing this to our attention @cliffordsnow.

I had a deeper look at the internals, especially at the method `getFerrySpeed` in `AbstractFlagEncoder`. 

The issue is, that we have to handle ferries as regular edges in our graph. For example for foot, the maxSpeed is 10, which is already a higher, just for ferries, so for Foot, we cannot create Edges with speeds higher than 10km/h. Also if the speed is inside the bounds of maxSpeed, the factor for the speedEncoder is different for car (5) to foot and bike (2). 

I just created a [commit](https://github.com/boldtrn/graphhopper/commit/b6584d07d9af2c1a05cb55ea1f512d676b8d9044) that would mostly fix the differences between car and bike, but due to the maxSpeed issue cannot do this for foot. I think this might be actually fixable once we merge #1112. Especially for the MappedDecimalEncodedValue, as we could map higher speeds for FootEncoder using that.

BTW: the above mentioned commit also fixes a bug in the getFerrySpeed method, even if we don't merge that commit, I think we should use the updtaed getFerrySpeed method (with the old parameters). There was a bug that we only set the [shortTripSpeed](https://github.com/graphhopper/graphhopper/blob/5cf7ef182f2d25a626dc40bb91e450cb4359ff0f/core/src/main/java/com/graphhopper/routing/util/AbstractFlagEncoder.java#L472), but not the longTripSpeed. I fixed this by simply returning the speed early, which works well.
